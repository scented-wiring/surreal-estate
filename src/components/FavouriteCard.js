import React, { useState, useEffect } from "react";
import "../styles/FavouriteCard.css";
import PropTypes from "prop-types";
import axios from "axios";
import Alert from "./Alert";

const FavouriteCard = ({ propertyListing, _id, removeFavourite }) => {
  const [data, setData] = useState([]);
  const [alert, setAlert] = useState([{ message: "", isSuccess: false }]);

  useEffect(() => {
    axios
      .get(`http://localhost:4000/api/v1/PropertyListing/${propertyListing}`)
      .then((response) => {
        setData(response.data);
      })
      .catch(() => {
        setAlert({
          message: "Could not retrieve property data",
          isSuccess: false,
        });
      });
  }, [propertyListing]);

  return (
    <div className="favourite-card">
      <Alert message={alert.message} success={alert.isSuccess} />
      <div className="favourite-card-text">
        {data.title}, {data.city}
      </div>
      <button className="favourite-remove" onClick={() => removeFavourite(_id)}>
        Remove
      </button>
    </div>
  );
};

FavouriteCard.propTypes = {
  propertyListing: PropTypes.string.isRequired,
  _id: PropTypes.string.isRequired,
  removeFavourite: PropTypes.func.isRequired,
};

export default FavouriteCard;
