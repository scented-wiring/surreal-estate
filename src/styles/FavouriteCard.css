.favourite-card {
  background-color: #fff6e7;
  font-size: 16px;
  display: flex;
  justify-content: space-between;
  width: 500px;
  margin: 10px 0 0 0;
  padding: 10px;
  border: 1px solid black;
}

.favourite-remove {
  color: white;
  font-size: 16px;
}

.favourite-remove:hover {
  transition: all 0.4s ease 0s;
}
