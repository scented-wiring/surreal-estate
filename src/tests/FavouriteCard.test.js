import React from "react";
import { render } from "@testing-library/react";
import FavouriteCard from "../components/FavouriteCard";

describe("FavouriteCard", () => {
  it("renders a component", () => {
    const { asFragment } = render(<FavouriteCard />);
    expect(asFragment).toMatchSnapshot();
  });
});
