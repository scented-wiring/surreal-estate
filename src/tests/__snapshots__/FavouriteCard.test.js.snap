// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteCard renders correctly 1`] = `[Function]`;
